src/historywork/DownloadBucketsWork.cpp

```diff
@@ -70,13 +70,42 @@ DownloadBucketsWork::yieldMoreWork()
     auto hash = *mNextBucketIter;
     FileTransferInfo ft(mDownloadDir, HISTORY_FILE_TYPE_BUCKET, hash);
     auto w1 = std::make_shared<GetAndUnzipRemoteFileWork>(mApp, ft, mArchive);
+
+    OnFailureCallback cb = nullptr;
+    if (!mArchive)
+    {
+        auto getFileWeak = std::weak_ptr<GetAndUnzipRemoteFileWork>(w1);
+        cb = [getFileWeak, hash]() {
+            auto getFile = getFileWeak.lock();
+            if (getFile)
+            {
+                auto ar = getFile->getArchive();
+                if (ar)
+                {
+                    CLOG(INFO, "History") << fmt::format(
+                        "Bucket {} from archive {}", hash, ar->getName());
+                }
+            }
+        };
+    }
+
     auto w2 = std::make_shared<VerifyBucketWork>(
-        mApp, mBuckets, ft.localPath_nogz(), hexToBin256(hash));
+        mApp, mBuckets, ft.localPath_nogz(), hexToBin256(hash), cb);
     std::vector<std::shared_ptr<BasicWork>> seq{w1, w2};
     auto w3 = std::make_shared<WorkSequence>(
         mApp, "download-verify-sequence-" + hash, seq);
 
     ++mNextBucketIter;
     return w3;
 }
+
+void
+DownloadBucketsWork::onFailureRaise()
+{
+    if (mArchive)
```

Comment:
there is some weird asymmetry here between the `mArchive` and `!mArchive` errors:
in the `mArchive` mode, we always log this, where as in the other case we only log from this Work during a verification failure .

I think it makes more sense to instead change `cb` above to handle both cases: as `cb` is supposed to give context for the failure. As `cb` is called only when verifying, we may not even need special casing.